test/evmc/mocked_host.hpp

```diff
@@ -92,15 +92,10 @@ class MockedHost : public Host
         /// The address of the beneficiary account.
         address beneficiary;
 
-        /// The balance of the self-destructed account.
-        uint256be balance;
```

Comment:
JFYI: This is part of fuzzer state.

Reply:
Is it used for anything? The balance of self-destructed account should be 0, since it has been transferred away. Also you can read it from the account state if needed.

Reply:
Ah I see. May be I misunderstood this. My reasoning was as follows: Let's suppose balance is `1 gwei` and the account self destructs, then this balance of `1 gwei` should be recorded as the balance just before self destruct was called.

The motivation is that if there is a bug somewhere that makes balances change (e.g., legacy optimiser does not optimise a call hence leading to more gas being consumed which in turn reflects in the self balance), we can detect it by recording balance before self destruct for those test cases containing a self destruct.

Perhaps this is just poorly explained/not explained at all in my original PR that added the lines removed here :(

Reply:
@axic I think the builds will pass once we decide whether to keep balance in self destruct or not :-)

> The balance of self-destructed account should be 0

iiuc, we don't decrement the balance post self destruct. So perhaps we should retain balance in selfdestruct just to check if balance during selfdestruct between unoptimised and optimised code matches?

Reply:
One can always check the account balance of the selfdestructed and the beneficiary account. Shouldn't the fuzzer just dump those two values instead?

Reply:
> One can always check the account balance of the selfdestructed and the beneficiary account.

Post selfdestruct, the account is deleted, right? So I added a log to record the balance of the self destructed account just before the account is deleted.

Reply:
If you look at `EVMHost::selfdestruct()`, it should be clear that the entire balance must be at the beneficiary, so checking that should be enough?

Also it seems it is broken, because it should increment it, not overwrite. Will fix that separately.

Reply:
I see, the problem though is that the fuzzer may synthesize a beneficiary which does not exist :(

I'm guessing if the beneficiary address is not yet in the EVMHost state, then this won't work? Or does any non-existent account access actual insert into the map? May be it's the latter?

Reply:
`EVMHost::selfdestruct()` creates the beneficiary, just as it happens in the real chain.